src/components/EditProfileComponenets/UserDetails.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+
+import { updateUserDocument } from "@/lib/firebase/controller";
+
+import { useUser } from "@/context/UserContext";
+
+function UserDetails() {
+    const { user } = useUser();
+    const [userData, setUserData] = useState({
+        displayName: "",
+        fullName: "",
+        location: "",
+        avatar: "",
+        interest: "",
+    });
+
+    const handleInputChange = (e) => {
+        const { name, value } = e.target;
+        setUserData((prevUserData) => ({
+            ...prevUserData,
+            [name]: value,
+        }));
+    };
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        updateUserDocument(user.uid, userData)
+            .then(() => {
+                //console.log("User document updated successfully!");
+                //console.log(userData);
+            })
+            .catch((error) => {
+                error;
+                // console.error("Error updating user document:", error);
+            });
+    };
+
+    return (
```

Comment:
why create so many event listners why not catch them all from the form element like i did previously
 ?
 it will reduce the number of states created and the component re-rendering i suppose §